File: Day37/MatrixAddition.java
public class MatrixAddition {
    public static void main(String[] args) {
        int[][] matrix1={
            {1,2,3},
            {4,5,6},
            {7,8,9}
        };
        int[][] matrix2={
            {10,20,30},
            {40,50,60},
            {70,80,90}
        };
        if (matrix1.length != matrix2.length  ||  matrix1[0].length != matrix2[0].length){
        System.out.print("Matrices cannot be added");
        return;
        }
        int[][] result = new int[matrix1.length][matrix1[0].length];
        for(int i =0; i<matrix1.length; i++){
            for(int j =0; j<matrix1[0].length; j++){
                result[i][j] = matrix1[i][j] + matrix2[i][j];
            }
        }
        System.out.println("Resultant Matrix: ");
        for(int[] row: result){
            for(int element : row){
                System.out.print(element + " ");
            }
            System.out.println();
        }
    }
}

File: Day37/matrix_addition.py
""" Function to add the matrices"""
def add_matrices(matrix1,matrix2):
    """Check if matrices can be added"""
    if len(matrix1) != len(matrix2)  or len(matrix1[0]) != len(matrix2[0]):
        raise ValueError("Matrices cannot be added")
    result = []
    for i in range(0,len(matrix1),1):
        row = []
        for j in range(0,len(matrix1[0]),1):
            row.append(matrix1[i][j] + matrix2[i][j])
        result.append(row)
    return result
def print_matrix(matrix):
    """Function to print the matrix"""
    for row in matrix:
        print(row)

matrix_1 = [ [1,2,3] , [4,5,6] , [7,8,9] ]
matrix_2 = [ [10,20,30] , [40,50,60] , [70,80,90] ]

result_matrix = add_matrices(matrix_1,matrix_2)
print("Matrix 1: ")
print_matrix(matrix_1)
print("Matrix 2: ")
print_matrix(matrix_2)
print("Resultant Matrix : ")
print_matrix(result_matrix)
